src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/JDBCEncryptionDecryptionTest.java

```diff
@@ -2244,22 +2234,69 @@ void testNumerics(SQLServerStatement stmt, String cekName, String[][] table, Str
         }
     }
 
-    SQLServerKeyVaultAuthenticationCallback authenticationCallback = new SQLServerKeyVaultAuthenticationCallback() {
-        // @Override
-        ExecutorService service = Executors.newFixedThreadPool(2);
+    @ParameterizedTest
```

Comment:
None of these new tests seem to actually test authenticationCallback .As far as I understand, driver applications will be able to implement  `SQLServerKeyVaultAuthenticationCallback.getAccessToken()` API in 2 ways:
- Using adal4j library, see the sample [here](https://docs.microsoft.com/en-us/sql/connect/jdbc/azure-key-vault-sample-version-7.0?view=sql-server-ver15).
- Using msal4j library, as implemented below.
Regardless of which method the application chooses to use, the driver needs to be able to handle both. Please add appropriate tests for both scenarios.